rebalance.py

```diff
@@ -335,24 +334,84 @@ def _name_total_gen_htmiss_prior_pdf(self):
     def _name_total_gen_htmiss_prior_slope(self):
         return 'gen_htmiss_prior_slope'
 
+    def _get_gen_htmiss_prior_file(self):
+        rfile = './input/htmiss_prior.root'
+        return r.TFile(rfile)
+
+    def _figure_out_ht_bin(self, gen_ht):
+        '''Given the GEN-HT of the event, figure out which HT bin it corresponds to.'''
+        prior_ht_bins = [
+            '100_to_300',
+            '300_to_500',
+            '500_to_700',
+            '700_to_900',
+            '900_to_1300',
+            '1300_to_2000',
+            '2000_to_5000',
+        ]
+        for ht_bin in prior_ht_bins:
+            ht_bin_split = ht_bin.split('_')
+            min_ht, max_ht = int(ht_bin_split[0]), int(ht_bin_split[-1]) 
+
+            if (gen_ht >= min_ht) and (gen_ht <= max_ht):
+                return ht_bin
+
+        raise RuntimeError(f'Could not figure out the HT bin for HT: {gen_ht:.3f}')
+
     def _build_gen_htmiss_prior(self):
-        slope_name = self._name_total_gen_htmiss_prior_slope()
-        slope_variable = r.RooRealVar(
-            slope_name,
-            slope_name,
-            -0.05,
-        )
-        self._wsimp(slope_variable)
-
-        prior_pdf_name = self._name_total_gen_htmiss_prior_pdf()
-        htmiss_variable = self.ws.function(self._name_partial_gen_htmiss_variable(direction='pt'))
-        prior_pdf = r.RooExponential(
-            prior_pdf_name,
-            prior_pdf_name,
-            htmiss_variable,
-            slope_variable
-        )
-        self._wsimp(prior_pdf)
+        '''Build gen HTmiss prior, extracted from the source file.'''
+        # Get the source file for prior distributions
+        prior_input_file = self._get_gen_htmiss_prior_file()
+
+        # Get the (already saved to WS) HTmiss and HT variables
+        gen_ht = self.ws.function(self._name_total_gen_ht_variable())
+        gen_htmiss = self.ws.function(self._name_partial_gen_htmiss_variable(direction='pt')).evaluate()
```

Comment:
it's confusing that gen_ht is a function object and gen_htmiss is just a float. let's keep that more consistent

Reply:
Actually, gen_htmiss is not used -> delete